build_system/seqan3-config-version.cmake

```
@@ -0,0 +1,48 @@
+# -----------------------------------------------------------------------------------------------------
+# Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+# Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+# This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+# shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+# -----------------------------------------------------------------------------------------------------
+
+# This file adds version support for `find_package(SEQAN3 3.1)`.
+
+# Note that seqan3-config.cmake can be standalone and thus SEQAN3_CLONE_DIR might be empty.
+find_path (SEQAN3_CLONE_DIR NAMES build_system/seqan3-config.cmake HINTS "${CMAKE_CURRENT_LIST_DIR}/..")
+find_path (SEQAN3_INCLUDE_DIR NAMES "seqan3/version.hpp" HINTS "${SEQAN3_CLONE_DIR}/include")
+
+# extract version from seqan3/version.hpp header
+file(STRINGS "${SEQAN3_INCLUDE_DIR}/seqan3/version.hpp" SEQAN3_VERSION_HPP REGEX "#define SEQAN3_VERSION_(MAJOR|MINOR|PATCH)")
+string(REGEX REPLACE "#define SEQAN3_VERSION_(MAJOR|MINOR|PATCH) " "" PACKAGE_VERSION "${SEQAN3_VERSION_HPP}")
+string(REGEX REPLACE ";" "." PACKAGE_VERSION "${PACKAGE_VERSION}")
+
+if(PACKAGE_VERSION VERSION_LESS PACKAGE_FIND_VERSION)
+  set(PACKAGE_VERSION_COMPATIBLE FALSE)
+else()
+
+  if(PACKAGE_VERSION MATCHES "^([0-9]+)\\.")
+    set(CVF_VERSION_MAJOR "${CMAKE_MATCH_1}")
+  endif()
+
+  if(PACKAGE_FIND_VERSION_MAJOR STREQUAL CVF_VERSION_MAJOR)
+    set(PACKAGE_VERSION_COMPATIBLE TRUE)
+  else()
+    set(PACKAGE_VERSION_COMPATIBLE FALSE)
+  endif()
+
+  if(PACKAGE_FIND_VERSION STREQUAL PACKAGE_VERSION)
+      set(PACKAGE_VERSION_EXACT TRUE)
+  endif()
+endif()
+
+# if the installed or the using project don't have CMAKE_SIZEOF_VOID_P set, ignore it:
+if("${CMAKE_SIZEOF_VOID_P}" STREQUAL "" OR "8" STREQUAL "")
```

You compare "8" with "" which is always false:

```suggestion
if("${CMAKE_SIZEOF_VOID_P}" STREQUAL "" OR FALSE)
```

---

Yeah true, I just used a script to first auto-generated this file and afterwards fill it with stuff.